docs/tokencredentials.md

```diff
@@ -0,0 +1,163 @@
+
+// or pass instance of X509Certificate2
+// ClientCertificateCredential clientCertificateCredential = new ClientCertificateCredential(tenantId, clientId, certificatePath);
+
+GraphServiceClient graphServiceClient = new GraphServiceClient(clientCertificateCredential, scopes);
+
+User me = await graphClient.Me.Request()
+                .GetAsync();
+```
+
+## 6. AuthorizationCodeCredential
+
+This credential class provides a similar use to the [Authorization code provider](https://github.com/microsoftgraph/msgraph-sdk-dotnet-auth#a-authorization-code-provider) with the use of a client secret and can be used as follows. The authorization code flow enables native and web apps to securely obtain tokens in the name of the user. 
+
+```cs
+string[] scopes = {"User.Read"};
+
+AuthorizationCodeCredential authorizationCodeCredential = new AuthorizationCodeCredential(tenantId, clientId,  clientSecret, authCode);
+
+GraphServiceClient graphServiceClient = new GraphServiceClient(authorizationCodeCredential, scopes);
+
+User me = await graphClient.Me.Request()
+                .GetAsync();
+```
+
+# Other TokenCredentials
+
+It is also useful to know that Azure.Identity provides other TokenCredentials that may be useful. These include.
+
+## 1. EnvironmentCredential
+
+This credential enables the used of defined environment variables to configure authentication. The environment varables used are as follows.
+
+AZURE_TENANT_ID -   The Azure Active Directory tenant(directory) ID.
+
+AZURE_CLIENT_ID -   The client(application) ID of an App Registration in the tenant.
+
+AZURE_CLIENT_SECRET	- A client secret that was generated for the App Registration.
+
+AZURE_CLIENT_CERTIFICATE_PATH - A path to certificate and private key pair in PEM or PFX format, which can authenticate the App Registration.
+
+AZURE_USERNAME -    The username, also known as upn, of an Azure Active Directory user account.
+
+AZURE_PASSWORD -    The password of the Azure Active Directory user account. Note this does not support accounts with MFA enabled.
+
+Based on the environment varialbe defined, this credential ultimately uses a `ClientSecretCredential` or `UsernamePasswordCredential` to perform the authentication using these details.
+```cs
+string[] scopes = {"User.Read"};
+
+EnvironmentCredential environmentCredential = new EnvironmentCredential();
+
+GraphServiceClient graphServiceClient = new GraphServiceClient(environmentCredential, scopes);
+
+User me = await graphClient.Me.Request()
+                .GetAsync();
+```
+
+## 2. ChainedTokenCredential
+
+This credential class provides a way to chain TokenCredential instances to be used in the attempt of token aquisiitions until one of the `getToken` methods of the TokenCredential instances returns a non-default AccessToken. An example is as shown below. In the example, the application will first attempt to check if it can get an AccessToken from the Environment and then use the interactive browser flow if it is unable to do so.
```

Comment:
```suggestion
This credential class provides a way to chain TokenCredential instances to be used in the attempt of token acquisitions until one of the `getToken` methods of the TokenCredential instances returns a non-default AccessToken. An example is as shown below. In the example, the application will first attempt to check if it can get an AccessToken from the Environment and then use the interactive browser flow if it is unable to do so.
```